SharedContent/cpp-winui-template/WinUI3TemplateCpp/MainWindow.xaml.cpp

```diff
@@ -0,0 +1,101 @@
+﻿#include "pch.h"
+#include "MainWindow.xaml.h"
+#if __has_include("MainWindow.g.cpp")
+#include "MainWindow.g.cpp"
+
+#include "microsoft.ui.xaml.window.h" 
+#include "SampleConfiguration.h"
+#endif
+
+using namespace winrt;
+using namespace Microsoft::UI::Xaml;
+
+namespace winrt::WinUI3TemplateCpp::implementation
+{
+    MainWindow::MainWindow()
+    {
+        InitializeComponent();
+
+        Title(winrt::WinUI3TemplateCpp::Settings::FeatureName);
+
+        HWND hwnd = GetWindowHandle();
+
+        LoadIcon(hwnd, L"windows-sdk.ico");
+        SetWindowSize(hwnd, 1220, 1080);
+        PlacementCenterWindowInMonitorWin32(hwnd);
+    }
+
+    HWND MainWindow::GetWindowHandle()
+    {
+        if (_hwnd == nullptr)
+        {
+            Window window = this->try_as<Window>();
```

Comment:
What does `try_as` do when it fails? If it returns a null reference, the app may crash on the next line with something less helpful than a cast syntax that throws when it fails.

Reply:
+1. Use `as<>` when you require a non-null result. Use `try_as<>` if you want to handle failure.

I've found this to be a popular anti-pattern: Using a special version that returns null, and then crashing on null anyway. If you are going to crash on null, then don't ask for the special version that returns null.

Reply:
Looks like `as<>`  doesn't exist

Reply:
fixed by QI directly with IWindowNative